### PR TITLE
MAINT: Bump to 0.13.4

### PR DESCRIPTION
**Pull request by larsoner**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] ~~Bumped the build number (if the version is unchanged)~~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #55

### CODE REVIEW

**Comment by larsoner** [comment]
@conda-forge-admin, please rerender

**Comment by larsoner** [comment]
I rolled the 3.11 transition into this but let me know if you'd prefer it be done separately

**Comment by larsoner** [comment]
@conda-forge-admin, please rerender

**Comment by bashtage** [comment]
I think the pypy won't build.

**Comment by larsoner** [comment]
I don't see any PyPy runs in the current CI run, they all end with `___cpython`. Maybe that's a good thing, then?

File: .azure-pipelines/azure-pipelines-win.yml [hunk 1]
```diff
@@ -8,21 +8,18 @@ jobs:
     vmImage: windows-2019
   strategy:
     matrix:
-      win_64_numpy1.19python3.7.____73_pypy:
-        CONFIG: win_64_numpy1.19python3.7.____73_pypy
```
Comment (larsoner) [inline .azure-pipelines/azure-pipelines-win.yml:12]: I guess these were removed as part of the migration, but maybe it's a bug

**Comment by larsoner** [comment]
Failures look real/related. Feel free to push commits or close. The autotick-bot should come through with a 0.13.3 update at some point, then a new PR with 3.11 added should follow after that I think

**Comment by ocefpaf** [comment]
Looks like statsmodels numpy requirements on Winows and py310 are: 

```
"numpy==1.22.3; python_version=='3.10' and platform_system=='Windows' and platform_python_implementation != 'PyPy'",
```


See https://github.com/statsmodels/statsmodels/blob/v0.13.3/pyproject.toml#L8

**Comment by bashtage** [comment]
I think cython needs to get bumped to 2.9.32.

**Comment by bashtage** [comment]
@ocefpaf those are build requirements, not run requirements.  It should build with anything newer too.

**Comment by larsoner** [comment]
I pushed a commit to ensure `cython >=0.29.32`, but in the failed build it looks like it used:
```
cython:           0.29.32-py311ha362b79_1    conda-forge
```
so I'm not optimistic about this fixing things

**Comment by ocefpaf** [comment]
> @ocefpaf those are build requirements, not run requirements. It should build with anything newer too.

You mean run with newer, right? The build requirement is pinned to numpy==1.22.3, and conda-forge there is using 1.23, so that is wrong.

**Comment by larsoner** [comment]
> The build requirement is pinned to numpy==1.22.3, and conda-forge there is using 1.23, so that is wrong.

To me this suggests that the `pin_compatible` for conda-forge is wrong, but I'm no expert here.

Also the failure looks perhaps unrelated to NumPy, more like Cython maybe?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=602745&view=logs&jobId=9a864fd9-6c8f-52ca-79ce-2aa6dca1a1de
```
  statsmodels/tsa/_innovations.c: In function '__Pyx_PyErr_GetTopmostException':
  statsmodels/tsa/_innovations.c:20057:21: error: '_PyErr_StackItem' {aka 'struct _err_stackitem'} has no member named 'exc_type'
  20057 |     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
        |                     ^~
  statsmodels/tsa/_innovations.c:20057:51: error: '_PyErr_StackItem' {aka 'struct _err_stackitem'} has no member named 'exc_type'
  20057 |     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
        |                                                   ^~
  statsmodels/tsa/_innovations.c: In function '__Pyx__ExceptionSave':
  statsmodels/tsa/_innovations.c:20071:21: error: '_PyErr_StackItem' {aka 'struct _err_stackitem'} has no member named 'exc_type'
  20071 |     *type = exc_info->exc_type;
        |                     ^~
  statsmodels/tsa/_innovations.c:20073:19: error: '_PyErr_StackItem' {aka 'struct _err_stackitem'} has no member named 'exc_traceback'
  20073 |     *tb = exc_info->exc_traceback;
        |                   ^~
  statsmodels/tsa/_innovations.c: In function '__Pyx__ExceptionReset':
  statsmodels/tsa/_innovations.c:20087:24: error: '_PyErr_StackItem' {aka 'struct _err_stackitem'} has no member named 'exc_type'
  20087 |     tmp_type = exc_info->exc_type;
...
```
Because it seems like this is Cython-generated from https://github.com/statsmodels/statsmodels/blob/main/statsmodels/tsa/_innovations.pyx

Did you have to do anything special with Cython to get it to compile in 3.11-compatible mode when making the wheels?

**Comment by larsoner** [comment]
> The build requirement is pinned to numpy==1.22.3, and conda-forge there is using 1.23, so that is wrong.

Also I think you're talking about 3.10 -- where conda-forge appears to use 1.21 actually -- and the builds are failing for Python 3.11, where the current pyproject.toml does not pin, right?

https://github.com/statsmodels/statsmodels/blob/65b785be1b9ecce8bfeb6ca2e00e2007473a3128/pyproject.toml#L11

**Comment by bashtage** [comment]
There's a bigger problem here. I have a feeling that the tarball was not built with the very latest scython and so the code in it is not compatible with Python 3.11. I'll probably need to issue a new point release later tonight to work around this issue.

**Comment by bashtage** [comment]
To be clear the issue only happens when building from the tar ball because the tarball contains the cythonized code translated to C.  When you build from the tarball you don't actually need cython at all. I think there's a workaround which is to set an environmental variable although I can't remember it off the top of my head.

**Comment by bashtage** [comment]
The wheels don't fail because they're generated from the GitHub repository rather than from the source tarball.

**Comment by ocefpaf** [comment]
> To me this suggests that the `pin_compatible` for conda-forge is wrong, but I'm no expert here.

Nope. Only that, if numpy 1.22 is also required on Windows for py311 we need to rebuild it for py311 first. Conda-forge usually bump the pin to latest numpy in migrations to avoid doing extra work.

> The wheels don't fail because they're generated from the GitHub repository rather than from the source tarball.

Yep. We need to re-cythonnize here to make it pass 3.11. However I need to take a closer look b/c it seems, [from this try/except block](https://github.com/statsmodels/statsmodels/blob/main/setup.py#L24-L44), that re-cythonization already happens if cython is present.

**Comment by bashtage** [comment]
The 0.13.4 release is up.  The 0.13.3 release showed Cython 0.29.25 in the C files.  The 0.13.4 C files show 0.29.32.

**Comment by bashtage** [comment]
Fixed now.  Thanks.

**Comment by larsoner** [comment]
Thanks for the quick fix @bashtage !

**Pull request merged**